### PR TITLE
feat: created new "Strinct" param.

### PR DESCRIPTION
**Pull request by krackerman**

### CODE REVIEW

File: src/params/strinct.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  'undefined',
+  'undef',
+  'null',
+  'NULL'
+];
+
+class Strinct extends Str {
+  constructor(...args) {
+    super(...args);
+    this.regex = new RegExp(`^(${rejectedStrings.join('|')})$`);
```
Comment (simlu) [inline src/params/strinct.js]: needs to be regex escaped

File: src/params/strinct.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const Str = require('./str');
+
+const rejectedStrings = [
+  '',
+  'undefined',
+  'undef',
+  'null',
+  'NULL'
```
Comment (simlu) [inline src/params/strinct.js]: add more as discussed

Comment (krackerman) [reply src/params/strinct.js]: Added more rejected strings from
 https://github.com/minimaxir/big-list-of-naughty-strings/blob/master/naughtystrings/internal/resource.go

File: src/params/strinct.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 const Str = require('./str');
 
+const escapeRegExp = arr => arr.map(s => s.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&'));
```
Comment (simlu) [inline src/params/strinct.js]: Shouldn't take an array. Should take a string

File: src/params/strinct.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  'None',
+  'NaN',
+  'Infinity',
+  '-Infinity',
+  'INF',
+  '"',
+  '\'',
+  '""',
+  '\'\'',
+  '"\'"',
```
Comment (simlu) [inline src/params/strinct.js]: Rm

File: src/params/strinct.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  '_',
+  '-',
+  '--',
+  'NUL'
+];
+
+class Strinct extends Str {
+  constructor(...args) {
+    super(...args);
+    this.regex = new RegExp(`^(${rejectedStrings.map(s => escapeRegExp(s)).join('|')})$`);
```
Comment (simlu) [inline src/params/strinct.js]: Almost good. But you forgot to negate it

**Pull request merged**